account_payment_order/models/account_payment_line.py

```diff
@@ -0,0 +1,124 @@
+    move_line_id = fields.Many2one(
+        'account.move.line', string='Journal Item')
+    ml_maturity_date = fields.Date(
+        related='move_line_id.date_maturity', readonly=True)
+    currency_id = fields.Many2one(
+        'res.currency', string='Currency of the Payment Transaction',
+        required=True,
+        default=lambda self: self.env.user.company_id.currency_id)
+    # v8 field : currency
+    amount_currency = fields.Monetary(
+        string="Amount", currency_field='currency_id')
+    amount_company_currency = fields.Monetary(
+        compute='compute_amount_company_currency',
+        string='Amount in Company Currency', readonly=True,
+        currency_field='company_currency_id')  # v8 field : amount
+    partner_id = fields.Many2one(
+        'res.partner', string='Partner', required=True,
+        domain=[('parent_id', '=', False)])
+    partner_bank_id = fields.Many2one(
+        'res.partner.bank', string='Partner Bank Account', required=False,
+        ondelete='restrict')  # v8 field : bank_id
+    date = fields.Date(string='Payment Date')
+    communication = fields.Char(
+        string='Communication', required=True,
+        help="Label of the payment that will be seen by the destinee")
+    communication_type = fields.Selection([
+        ('normal', 'Free'),
+        ], string='Communication Type', required=True, default='normal')
+    # v8 field : state
+    bank_line_id = fields.Many2one(
+        'bank.payment.line', string='Bank Payment Line', readonly=True)
+
+    _sql_constraints = [(
+        'name_company_unique',
+        'unique(name, company_id)',
+        'A payment line already exists with this reference '
+        'in the same company!'
+        )]
+
+    @api.model
+    def create(self, vals):
+        if vals.get('name', 'New') == 'New':
+            vals['name'] = self.env['ir.sequence'].next_by_code(
+                'account.payment.line') or 'New'
+        return super(AccountPaymentLine, self).create(vals)
+
+    @api.multi
+    @api.depends(
+        'amount_currency', 'currency_id', 'company_currency_id', 'date')
+    def compute_amount_company_currency(self):
+        for line in self:
+            if line.currency_id and line.company_currency_id:
+                line.amount_company_currency = line.currency_id.with_context(
+                    date=line.date).compute(
+                        line.amount_currency, line.company_currency_id)
+
+    @api.multi
+    def payment_line_hashcode(self):
+        self.ensure_one()
+        bplo = self.env['bank.payment.line']
+        values = []
+        for field in bplo.same_fields_payment_line_and_bank_payment_line():
+            values.append(unicode(self[field]))
+        hashcode = '-'.join(values)
+        return hashcode
+
+    @api.onchange('partner_id')
+    def partner_id_change(self):
+        partner_bank = False
+        if self.partner_id and self.partner_id.bank_ids:
```

Comment:
You can remove `self.partner_id and`.

If `self.partner_id` is `False`
then `self.partner_id.bank_ids` will return `False`